pkg/localkube/localkube.go

```diff
@@ -17,61 +17,90 @@ limitations under the License.
-	if len(args) < 1 {
-		return errors.New("you must choose start <name>, stop <name>, or status")
+func (lk LocalkubeServer) GetEtcdDataDirectory() string {
+	return path.Join(lk.LocalkubeDirectory, "etcd")
+}
+
+func (lk LocalkubeServer) GetDNSDataDirectory() string {
+	return path.Join(lk.LocalkubeDirectory, "dns")
+}
+
+func (lk LocalkubeServer) GetCertificateDirectory() string {
+	return path.Join(lk.LocalkubeDirectory, "certs")
+}
+func (lk LocalkubeServer) GetPrivateKeyCertPath() string {
+	return path.Join(lk.GetCertificateDirectory(), "apiserver.key")
+}
+func (lk LocalkubeServer) GetPublicKeyCertPath() string {
+	return path.Join(lk.GetCertificateDirectory(), "apiserver.crt")
+}
+
+func (lk LocalkubeServer) GetAPIServerSecureURL() string {
+	return fmt.Sprintf("https://%s:%d", lk.APIServerAddress.String(), lk.APIServerPort)
+}
+
+func (lk LocalkubeServer) GetAPIServerInsecureURL() string {
+	return fmt.Sprintf("http://%s:%d", lk.APIServerInsecureAddress.String(), lk.APIServerInsecurePort)
+}
+
+// Get the host's public IP address
+func (lk LocalkubeServer) GetHostIP() (net.IP, error) {
+	return utilnet.ChooseBindAddress(net.ParseIP("0.0.0.0"))
+}
+
+func (lk LocalkubeServer) GenerateCerts() error {
+
+	if util.CanReadFile(lk.GetPublicKeyCertPath()) && util.CanReadFile(lk.GetPrivateKeyCertPath()) {
+		fmt.Println("Using these existing certs: ", lk.GetPublicKeyCertPath(), lk.GetPrivateKeyCertPath())
+		return nil
 	}
 
-	switch args[0] {
-	case "start":
-		// check if just start
-		if len(args) == 1 {
-			fmt.Fprintln(out, "Starting LocalKube...")
-			lk.StartAll()
-			return nil
-		} else if len(args) == 2 {
-			serverName := args[1]
-			fmt.Fprintf(out, "Starting `%s`...\n", serverName)
-			return lk.Start(serverName)
-
-		} else {
-			return errors.New("start: too many arguments")
-		}
-	case "stop":
-		// check if just stop
-		if len(args) == 1 {
-			fmt.Fprintln(out, "Stopping LocalKube...")
-			lk.StopAll()
-			return nil
-		} else if len(args) == 2 {
-			serverName := args[1]
-			fmt.Fprintf(out, "Stopping `%s`...\n", serverName)
-			return lk.Stop(serverName)
-		}
-	case "status":
-		fmt.Fprintln(out, "LocalKube Status")
-		fmt.Fprintln(out, "################\n")
-
-		fmt.Fprintln(out, "Order\tStatus\tName")
-		for num, server := range lk.Servers {
-			fmt.Fprintf(out, "%d\t%s\t%s\n", num, server.Status(), server.Name())
-		}
-
-		fmt.Fprintln(out)
+	alternateIPs := []net.IP{lk.ServiceClusterIPRange.IP}
+	alternateDNS := []string{fmt.Sprintf("%s.%s", "kubernetes.default.svc", lk.DNSDomain), "kubernetes.default.svc", "kubernetes.default", "kubernetes"}
+	hostIP, err := lk.GetHostIP()
+	if err != nil {
+		fmt.Println("Failed to get host IP: ", err)
+		return err
 	}
-	return errors.New("you must choose start <name>, stop <name>, or status")
+
+	if err := utilcrypto.GenerateSelfSignedCert(hostIP.String(), lk.GetPublicKeyCertPath(), lk.GetPrivateKeyCertPath(), alternateIPs, alternateDNS); err != nil {
```

Comment:
+1, we should probably regenerate if the IP changes or if the file doesn't exist yet. After that this should be good to merge.